packages/@aws-cdk/aws-servicecatalog/lib/portfolio.ts

```diff
@@ -66,6 +70,24 @@ export interface IPortfolio extends cdk.IResource {
    * @param options Options for the initiate share
    */
   shareWithAccount(accountId: string, options?: PortfolioShareOptions): void;
+
+  /**
+   * Associate portfolio with the given product.
+   * @param product A service catalog produt.
+   */
+  addProduct(product: IProduct): void;
+
+  /**
+   * Associate Tag Options.
+   * A TagOption is a key-value pair managed in AWS Service Catalog.
+   * It is not an AWS tag, but serves as a template for creating an AWS tag based on the TagOption.
+   */
+  addTagOptions(tagOptions: TagOption): void;
+
+  /**
+   * Add a Resource Update Constraint.
+   */
+  addResourceUpdateConstraint(product: IProduct, options?: TagUpdatesOptions): void;
```

Comment:
See comment above - I'm not sure all these methods should be on `Portfolio`, and also the names could be improved I think.